src/tasks/models/tasksManager.ts

```diff
@@ -42,7 +42,11 @@ export class TasksManager {
       throw new NotFoundError(`Task ${taskId} not found`);
     }
     if (task.status === OperationStatus.FAILED) {
-      await this.failJob(task.jobId);
+      if (this.jobDefinitions.suspendingTaskTypes.includes(task.type)) {
```

Comment:
Its nice, you can use ternary condition instead of double if, seems nice and clean:
![image](https://github.com/user-attachments/assets/f147e725-de82-4107-bad6-5fb83a33f6cb)

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_operator